toolchain/check/import_ref.cpp

```diff
@@ -1429,6 +1429,41 @@ static auto TryResolveTypedInst(ImportRefResolver& resolver,
        .entity_name_id = entity_name_id});
 }
 
+static auto TryResolveTypedInst(ImportRefResolver& resolver,
+                                SemIR::BoundMethod inst) -> ResolveResult {
+  CARBON_CHECK(resolver.import_types().GetInstId(inst.type_id) ==
+               SemIR::BoundMethodType::SingletonInstId);
+  auto object_id = GetLocalConstantInstId(resolver, inst.object_id);
+  auto function_id = GetLocalConstantInstId(resolver, inst.function_id);
+
+  if (resolver.HasNewWork()) {
+    return ResolveResult::Retry();
+  }
+
+  return ResolveAs<SemIR::BoundMethod>(
+      resolver, {.type_id = resolver.local_context().GetSingletonType(
```

Comment:
For my own education, how come the `BoundMethod` is different from `Call` and uses a singleton id in its `type_id`? From what I can discern, `Call` has the return type of the call in its `type_id`?

Reply:
A `BoundMethod` is a thing that can be called (`x.Method`), but it has not yet been called, so its value and type reflect "callable thing" rather than "result of calling".